### PR TITLE
TST: test suggested in gh1758

### PR DESCRIPTION
**Pull request by andyfaff**
Adds the test mentioned in #1758.

### CODE REVIEW

File: doc/Makefile [hunk 1]
```diff
@@ -7,7 +7,7 @@
 # Use explicit "version_info" indexing since make cannot handle colon characters, and
 # evaluate it now to allow easier debugging when printing the varaible
 
-PYVER:=$(shell python3 -c 'from sys import version_info as v; print("{0}.{1}".format(v[0], v[1]))')
+PYVER:=$(shell python3.7 -c 'from sys import version_info as v; print("{0}.{1}".format(v[0], v[1]))')
```
Comment (rgommers) [inline doc/Makefile]: this won't work for everyone, but neither did the old code (clearly not for you). so let's just merge this, and fix it properly at some point by the thing numpy did here (I _think_ that works now in master, but it was a major pain to get right)

**Comment by rgommers** [comment]
thanks @andyfaff

**Comment by andyfaff** [comment]
@rgommers, the change made to the `scipy/doc/Makefile` was accidental, whilst investigating something else. I didn't mean for that to be included here. Should this be reverted?

**Comment by rgommers** [comment]
meh either way, we need to fix it properly anyway

**Comment by mdhaber** [comment]
@rgommers was there supposed to be a link? NumPy's makefile looks the same as ours did at the top.
`PYVER:=$(shell python3 -c 'from sys import version_info as v; print("{0}.{1}".format(v[0], v[1]))')`

Can we just make it a command line argument?

**Comment by rgommers** [comment]
Okay, follow-up in gh-10715.

> Can we just make it a command line argument?

that makes it less user-friendly, just `make html` should always succeed for a normal dev setup

**Comment by pv** [comment]
It is a command-line argument, e.g. `make PYVER=3.7` overrides it.

**Comment by peterbell10** [comment]
> `make html PYTHON=python3.8` overrides it.

`PYVER` is used in other places as well so this might not work right.

**Pull request merged**